src/tests/conformance/messages.rs

```diff
@@ -473,83 +501,76 @@ async fn eagerly_crawls_network_for_peers() {
     //  4. Send set of peer listener node addresses
     //  5. Expect the node to connect to each peer in the set
     //
-    // This test currently fails for zcashd; zebra fails (with a caveat).
+    // zcashd: Has different behaviour depending on connection direction.
+    //         If we initiate the main connection it sends Ping, GetHeaders,
+    //         but never GetAddr.
+    //         If the node initiates then it does send GetAddr, but it never connects
+    //         to the peers.
     //
-    // Current behaviour:
-    //
-    //  zcashd: Has different behaviour depending on connection direction.
-    //          If we initiate the main connection it sends Ping, GetHeaders,
-    //          but never GetAddr.
-    //          If the node initiates then it does send GetAddr, but it never connects
-    //          to the peers.
-    //
-    // zebra:   Fails, unless we keep responding on the main connection.
-    //          If we do not keep responding then the peer connections take really long to establish,
-    //          failing the test completely.
+    // zebra:  Fails, unless we keep responding on the main connection.
+    //         If we do not keep responding then the peer connections take really long to establish,
+    //         failing the test completely.
     //
-    //          Related issues: https://github.com/ZcashFoundation/zebra/pull/2154
-    //                          https://github.com/ZcashFoundation/zebra/issues/2163
+    //         Related issues: https://github.com/ZcashFoundation/zebra/pull/2154
+    //                         https://github.com/ZcashFoundation/zebra/issues/2163
 
-    // create tcp listeners for peer set (port is only assigned on tcp bind)
-    let mut listeners = Vec::new();
-    for _ in 0u8..5 {
-        listeners.push(TcpListener::bind(new_local_addr()).await.unwrap());
+    // Spin up a node instance.
+    let mut node: Node = Default::default();
+    node.initial_action(Action::WaitForConnection).start().await;
+
+    // Create 5 synthetic nodes.
+    const N: usize = 5;
+    let mut synthetic_nodes = Vec::with_capacity(N);
+    for _ in 0..N {
+        let synthetic_node = SyntheticNode::new(SyntheticNodeConfig {
+            enable_handshaking: true,
+            message_filter: MessageFilter::with_all_auto_reply(),
+            ..Default::default()
+        })
+        .await
+        .unwrap();
+
+        synthetic_nodes.push(synthetic_node);
     }
 
-    // get list of peer addresses
-    let peer_addresses = listeners
+    // Collect their addrs.
+    let addrs = synthetic_nodes
         .iter()
-        .map(|listener| listener.local_addr().unwrap())
+        .map(|node| node.listening_addr())
+        .map(|addr| NetworkAddr::new(addr))
```

Comment:
Clippy - redundant closure
```sugggest
.map(NetworkAddr::new)
```

Reply:
Ended up condensing the two `map` calls, good catch—CI didn't see it.